### DISCUSSION TITLE
Chocolatey GUI modify open source file

### DISCUSSION THREAD

**Discussion by Timmy5818** [chocolatey/ChocolateyGUI#940] [Q&A]
I follow the official documentation to set up the server.
I already created my chocolatey own source, and put some packages on it.
This is my Chocolatey source :
[Chocolatey source](https://chocolateypowershell.azurewebsites.net/)

I'm wondering if I could use Chocolatey GUI to be an interface, and people can login and download apps from my source.
Here is my idea, I'll modify the GUI open source, and make a login page, add my Chocolatey source only and pack it as a .exe file, finally post on my website to let everyone download.

In GUI part, I won't push it to community, just for self use.
But I don't know which part should I go first because I'm not quite understand the project. If I'll like to create login page first, where can I add it in project?

This is my current problem now,  I'm not sure can this goal achieve or not.

**Comment by pauby** [comment]
This was recently discussed on Discord starting [here](https://discord.com/channels/778552361454141460/897088817293574154/973761271839227924) and ending [here](https://discord.com/channels/778552361454141460/897088817293574154/974523636578914314).

**Reply by Timmy5818** [reply]
> This was recently discussed on Discord starting [here](https://discord.com/channels/778552361454141460/897088817293574154/973761271839227924) and ending [here](https://discord.com/channels/778552361454141460/897088817293574154/974523636578914314).

Actually that question on discord was from me. I ask the question again here because I thought they probably forget to respond
to me, or other reasons else. Another reason is I want to know different developers' perspective about this goal.

**Comment by pauby** [comment]
> I'm wondering if I could use Chocolatey GUI to be an interface, and people can login and download apps from my source.
Here is my idea, I'll modify the GUI open source, and make a login page, add my Chocolatey source only and pack it as a .exe file, finally post on my website to let everyone download.

This question was answered. We don't have anything like that in Chocolatey GUI, and you'd have to build that yourself.

> In GUI part, I won't push it to community, just for self use.
But I don't know which part should I go first because I'm not quite understand the project. If I'll like to create login page first, where can I add it in project?

This question we can't answer. You're asking us to evaluate your proposal and tell you how to do it. That's for you to determine. This is an open-source project where you are free, within the license, to modify and use it. If you have a specific question around something, then that we may be able to answer that, but we can't tell you how to build your own solution.

We provide community support and help for Chocolatey GUI and how it functions as a product. We can't provide support and help for something you want to go off and build.

> Another reason is I want to know different developers' perspective about this goal.

If you are looking for those different perspectives on your own project, then this isn't the place to do it. This repository is for the Chocolatey GUI product.

I appreciate some of this is a fine line. What we don't want is discussions about a commercial solution that you're building. That is going to cause confusion, and that's not what it's here for. That would be something you would build and gather those developers to give you that perspective.

**Reply by Timmy5818** [reply]
I totally understand. Thanks for your helping. Actually this is my first time using open source, so I might 
not be aware of this kind of problem. I'm so sorry for that.

**Reply by pauby** [reply]
There is no need to apologise. We all have to learn. Welcome to the huge, fascinating and sometimes difficult world of open-source 😄
